From d4bcd004a6ff49bb195ee9152e82dcb97a875b28 Mon Sep 17 00:00:00 2001
From: Ricard Wanderlof <polluxsynth@butoba.net>
Date: Mon, 21 Mar 2022 00:49:26 +0100
Subject: [PATCH 1839/1936] SVFilter.cpp: Avoid clicks when UI updates freq, Q
 or gain (#348)

In the setstages() method, only call cleanup() when needed,
i.e. when the number of stages is actually changed (same as
in AnalogFilter.cpp).

(The reason this is a problem is because setstages() is
called whenever the frequency, Q, filter tracking or gain
is changed in the UI).
---
 src/DSP/SVFilter.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/DSP/SVFilter.cpp b/src/DSP/SVFilter.cpp
index e6e66123..108de945 100644
--- a/src/DSP/SVFilter.cpp
+++ b/src/DSP/SVFilter.cpp
@@ -151,9 +151,11 @@ void SVFilter::setstages(int stages_)
 {
     if(stages_ >= MAX_FILTER_STAGES)
         stages_ = MAX_FILTER_STAGES - 1;
-    stages = stages_;
-    cleanup();
-    computefiltercoefs();
+    if (stages != stages_) {
+        stages = stages_;
+        cleanup();
+        computefiltercoefs();
+    }
 }
 
 float *SVFilter::getfilteroutfortype(SVFilter::fstage &x) {
-- 
2.47.0

